app/views/play/cinematic/DialogSystem.js

```diff
@@ -0,0 +1,314 @@
+  }
+}
+
+/**
+ * @typedef {Object} SvgBubbleOptions
+ * @property {number} x - x position of the bubble.
+ * @property {number} y - y position
+ * @property {number} width
+ * @property {number} height
+ * @property {'left'|'right'} side
+ * @property {string} class - the unique class for this bubble
+ */
+
+/**
+ * Returns a svg speech bubble that can be attached to an svg canvas.
+ * @param {SvgBubbleOptions} svgOptions - Svg options for generating the svg shape.
+ * @returns {SVGGElement} the group element of the svg bubble.
+ */
+function createSvgShape ({ x, y, width, height, side, clazz }) {
+  const g = document.createElementNS(SVGNS, 'g')
+  g.setAttribute('transform', `translate(${x - padding}, ${y + height + 1 - padding})`)
+  g.setAttribute('fill', 'white')
+  // Animation will reveal this svg.
+  g.setAttribute('opacity', '0')
+  g.setAttribute('class', clazz)
+
+  const rect = document.createElementNS(SVGNS, 'rect')
+  rect.setAttribute('width', `${width}`)
+
+  // Animation will add the future height.
+  rect.setAttribute('height', `${0}`)
+
+  const path = document.createElementNS(SVGNS, 'path')
+  path.setAttribute('d', 'M -20 20 l 21 -10 0 20 z')
+  if (side === 'right') {
+    path.setAttribute('transform', `translate(${width},0) scale(-1, 1)`)
+  }
+  g.appendChild(rect)
+  g.appendChild(path)
+  return g
+}
+
+/**
+ * Extract the text from the DialogNode, and transform it into an html element ready for animation.
+ *
+ * This function handles internationalization, text interpolation and html processing.
+ *
+ * We use very light weight [Javascript-Templates](https://blueimp.github.io/JavaScript-Templates/)
+ * in order to provide text templating.
+ *
+ * @param {DialogNode} dialogNode
+ * @param {Object} context - The object referred to as `o` in the text templates.
+ * @param {bool} wrap - Whether we want to wrap the transpiled text in html tags.
+ * @returns {HTMLElement|undefined} The processed element.
+ */
+export function processText (dialogNode, context, wrap = true) {
+  let text = utils.i18n(dialogNode, 'text')
+  text = tmpl(text || '', context)
+  if (!text) {
+    return undefined
+  }
+
+  if (wrap) {
+    return wrapText(`<div>${text}</div>`)
+  }
+  return text
+}
+
+/**
+ * From the stackoverflow answer: https://stackoverflow.com/a/20693791/6421793
+ *
+ * Default behaviour:
+ *
+ * Wraps each letter with a span set at opacity 0.
+ * This lets us calculate the bounding box of the html element.
+ * To prevent the possible occurrance of line breaks in the middle of a word
+ * we must also wrap words with a span tag.
+ *
+ * @param {string} htmlString is a raw string representation of html
+ * @param {function} wrapLetterString takes a letter and can return a wrapped letter.
+ * @param {function} wrapWordString takes a word and can return a wrapped word.
+ * @returns {string} transformed html string.
+ */
+export function wrapText (htmlString, wrapLetterString, wrapWordString) {
```

Comment:
Why are we exporting processText and wrapText, can we move them to a common place like `utils` if we want them to be re-usable?

Reply:
I agree and really don't like these exports, so I'll give my problem hoping for a solution.

I wouldn't want to move these files to `utils` as they're very specific to the Dialog System. Specifically the `wrapText` function is very specific wrapping text in individual tags.
However these are also not easy functions so I want to ensure they have some test coverage.

A test that tests public behavior of the DialogSystem is difficult due to testing side effects. 
Thus I'm exporting these for testing to ensure their behavior is predictable for the system depending on them.

Interested in other tactics to solve the above problem.

Reply:
How about a common file thats specific to dialog system like `dialogSystemHelper`

Reply:
hmmmm. The more I think about that the more I like it.
I see keeping the functions in `dialogSystem.js` as one option as they're very related.
But then they are exported muddying up with public exports of `dialogSystem`. Whilst placing them in a `dialogSystemHelper` file would allow them to be exported but it makes their intent clearer.
I'm a fan of the `dialogSystemHelper` idea 🚀 👏

Reply:
This is also excellent because the `dialogsystem.js` file currently contains 2 ideas. The idea of drawing html nodes on a div canvas and the idea of drawing svgs onto an svg canvas.

These could both be split into separate files and the DialogSystem just uses them.
Not doing the full refactor in  this PR. But creating the initial folder and moving out the functions into the dialogSystemHelper file.